docs/lsp/howto/migrate-to-v1.rst

```diff
@@ -0,0 +1,76 @@
+Instead, you should now have a single, global installation that can be reused across projects.
+We recommend that you use `pipx <https://pipx.pypa.io/stable/>`__ to manage this installation for you::
+
+   $ pipx install esbonio   # Installs esbonio globally in an isolated environment
+   $ pipx upgrade esbonio   # Upgrade esbonio and its dependencies
+
+.. note::
+
+   If you use the Esbonio VSCode extension, the language server is automatically included as part of the extension itself, no separate installation required!
+
+Configuration Changes
+---------------------
+
+.. tip::
+
+   With the release of ``v1.x``, Esbonio's configuration system has been overhauled, see :ref:`lsp-configuration` for all of the available configuration options and methods.
+
+While ``esbonio`` can now be installed globally, it still needs access to your project's development environment in order to properly understand it.
+This means the two most imporant configuration values are
+
+- :esbonio:conf:`esbonio.sphinx.pythonCommand`: For telling ``esbonio`` the command it needs to run in order to use the correct Python environment.
+- :esbonio:conf:`esbonio.sphinx.buildCommand`: For telling ``esbonio`` the ``sphinx-build`` command you use to build your documentation. This is so that the server invoke Sphinx with the correct arguments.
+
+The following table outlines the configuration options that have been removed in ``v1.x`` and what their correpsonding replacement is
+
++-----------------------------------------+-------------------------------------------------+-------------+
+| Removed Option                          | Replacement                                     | Notes       |
++=========================================+=================================================+=============+
+| - ``esbonio.server.hideSphinxOutput``   | :esbonio:conf:`esbonio.sphinx.buildCommand`     |             |
+| - ``esbonio.sphinx.buildDir``           |                                                 |             |
+| - ``esbonio.sphinx.builderName``        |                                                 |             |
+| - ``esbonio.sphinx.confDir``            |                                                 |             |
+| - ``esbonio.sphinx.doctreeDir``         |                                                 |             |
+| - ``esbonio.sphinx.forceFullBuild``     |                                                 |             |
+| - ``esbonio.sphinx.keepGoing``          |                                                 |             |
+| - ``esbonio.sphinx.makeMode``           |                                                 |             |
+| - ``esbonio.sphinx.numJobs``            |                                                 |             |
+| - ``esbonio.sphinx.quiet``              |                                                 |             |
+| - ``esbonio.sphinx.silent``             |                                                 |             |
```

Comment:
Not sure if you're planning on re-implementing `silent`, but it is handy for de-duplicating diagnostics when also using rstcheck (rstcheck works outside of Sphinx projects).

Reply:
Each of the removed config options should correspond with a cli option you can pass to `sphinx-build`. For example, there is the [`-Q`](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-Q) flag which should do the equivalent to what `esbonio.sphinx.silent` did. (Feel free to open an issue if it does not!)

I should probably update the guide pointing out the equivalent `sphinx-build` flag for each of the removed options